cmd/client_test.go

```diff
@@ -0,0 +1,139 @@
+func generateRedisProbeDef(name string) *probes_configpb.ProbeDef{
+        probeDef := &probes_configpb.ProbeDef{ // Create probe def
+                Name: proto.String(name),
+                Type: probes_configpb.ProbeDef_EXTENSION.Enum(),
+                Targets: &targetspb.TargetsDef{
+                        Type: &targetspb.TargetsDef_DummyTargets{},
+                },
+        }
+
+	// Add RedisProbe extension to probeDef
+        op := myprobe.ProbeConf_Op.Enum(myprobe.ProbeConf_SET)
+        proto.SetExtension(probeDef, myprobe.E_RedisProbe, &myprobe.ProbeConf{Op: op, Key: proto.String("testkey"), Value: proto.String("testval")})
+        return probeDef
+}
+
+// TestRegisterAndAddProbe tests that an extension probe type can be registered and added to Cloudprober without error.
+func TestRegisterAndAddProbe(t *testing.T) {
+	// Create a probe and then register and add it to the prober
+	probeName1 := "testExtension1"
+	RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName1), &myprobe.Probe{})
+
+	// Check if the probe was added correctly
+	resp := ListProbes()
+	respProbes := resp.GetProbe()
+
+	if respProbes[0].GetName() != probeName1 {
+		t.Errorf("Expected probe %s to be running, got probe %s running", probeName1, respProbes[0].GetName())
+	}
+
+	RemoveProbe(probeName1)
+
+	// Try adding a second probe to see if the registration is still in place
+	probeName2 := "testExtension2"
+	RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName2), &myprobe.Probe{})
+
+	resp = ListProbes()
+	respProbes = resp.GetProbe()
+
+	if respProbes[0].GetName() != probeName2 {
+		t.Errorf("Expected probe %s to be running, got probe %s running", probeName2, respProbes[0].GetName())
+	}
+
+	RemoveProbe(probeName2)
+}
+
+// TestRemoveProbe tests that the RemoveProbe() method removes a probe from Cloudprober
+func TestRemoveProbe(t *testing.T) {
+	// Create a probe and then register and add it to the prober
+        probeName1 := "testExtension1"
+	RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName1), &myprobe.Probe{})
+
+	// Remove probe
+	RemoveProbe(probeName1)
+
+	// Verify that the probe was removed
+	resp := ListProbes()
+	respProbes := resp.GetProbe()
+	if len(respProbes) != 0 {
+		t.Errorf("Expected no probes to be running, got %d probes running", len(respProbes))
+		t.Error("Probes running: ", respProbes)
+	}
+}
+
+// TestListProbes tests that the ListProbes() method returns all active probes in Cloudprober
+func TestListProbes(t *testing.T) {
+	// Create a probe and then register and add it to the prober
+	probeName1 := "testExtension1"
+	RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName1), &myprobe.Probe{})
+
+	// Check if the probe was added correctly
+	resp := ListProbes()
+	respProbes := resp.GetProbe()
+
+	if respProbes[0].GetName() != probeName1 {
```

Comment:
nit: Just to keep comparisons "parallel" in nature, start your set of test-case variables with a "0" suffix. e.g. probeName0, probeName1, etc. This way, when you compare respProbes[0] you can compare it with probeName0.

Reply:
That's a good practise, thanks, I'll make these changes and keep this in mind.